beacon_node/eth1_chain/src/types.rs

```diff
@@ -0,0 +1,47 @@
+use parking_lot::RwLock;
+use std::collections::BTreeMap;
+use std::marker::{Send, Sync};
+use std::sync::Arc;
+use types::DepositData;
+use web3::futures::Future;
+use web3::types::*;
+
+/// Interface for getting Eth1 chain data.
+pub trait Eth1DataFetcher: Send + Sync + Clone {
+    /// Get block_header of the head of the chain.
+    fn get_current_block_number(&self) -> Box<dyn Future<Item = U256, Error = ()> + Send>;
+
+    /// Get block_hash at given height.
+    fn get_block_hash_by_height(
+        &self,
+        height: u64,
+    ) -> Box<dyn Future<Item = Option<H256>, Error = ()> + Send>;
```

Comment:
Do these also need to be box'd or can just impl Future?

Reply:
Same as above.